jaxb-ri/xjc/src/main/java/com/sun/tools/xjc/addon/accessors/PluginImpl.java

```diff
@@ -71,6 +89,58 @@ public boolean run( Outline model, Options opt, ErrorHandler errorHandler ) {
                     Logger.getLogger(PluginImpl.class.getName()).log(Level.SEVERE, null, ex);
                 }
             }
+            Iterator<JFieldVar> flds = co.implClass.fields().values().iterator();
+            while (flds.hasNext()) {
+            	try {
+            		JFieldVar fld = flds.next();
+            		String fldName = fld.name();
+            		Set<JAnnotationUse> annotsCopy = new HashSet<JAnnotationUse>(fld.annotations());
+            		String getterName = "get" + fldName.substring(0, 1).toUpperCase() + fldName.substring(1);
+            		String escapedName = null;
+            		for (JAnnotationUse annot : annotsCopy) {
+            			 Field clazzField = annot.getClass().getDeclaredField("clazz");
+                         clazzField.setAccessible(true);
+                         JClass cl = (JClass) clazzField.get(annot);
+                         if (cl.equals(model.getCodeModel()._ref(XmlElement.class))) {
+                        	 if (annot.getAnnotationMembers().containsKey("name")) {
+                        		 JAnnotationValue val = (JAnnotationValue) annot.getAnnotationMembers().get("name");
+                        		 StringWriter sw = new StringWriter();
+                        		 JFormatter ft = new JFormatter(sw); 
+                        		 val.generate(ft);
+                        		 String quotedName = sw.getBuffer().toString();
+                        		 String name = quotedName.substring(quotedName.indexOf('"') + 1, quotedName.lastIndexOf('"'));
+                        		 escapedName = NameConverter.standard.toClassName(name);
+                        		 getterName = "get" + escapedName;
+                        	 }
+                         }
+            		}
+            		
+            		JMethod getterMethod = co.implClass.getMethod(getterName, new JType[] {});
+            		if (getterMethod != null) {
+            			for (JAnnotationUse annot : annotsCopy) {
+            				fld.removeAnnotation(annot);
+            				getterMethod.annotate(annot);
+            			}
+            			String properGetterName = "get" + fldName.substring(0, 1).toUpperCase() + fldName.substring(1);
+            			if (!properGetterName.equals(getterName)) {
```

Comment:
What is the reason for normalizing names? When does it happen that field / getter does not match by javabean convention?

Reply:
My point is not about javabean convention, it is why there is that particular code. First property name is read from field name. Second property name is reread from XmlElement("name") value (if present). Than method is looked up by this name, and if found, name is recalculated according to field name and rewritten. Lets consider following example:
```java
@XmlElement(name = "abc")
private Long id;
public getId();
public void setId(Long id);
```
 - getterName is first set to "getId" by the field name + getPrefix
 - getterName is reset to `name` from `XmlElement` annotation and is now `getabc` (uppercasing 'a' looks to be omitted here).
 - next we are looking for `getabc` which is not present in the class and the code actually exits without doing anyting for `id` field.

Following will happen after, only when XmlElement(name) is not set or is set to the field name: 
 - Found getter by `getId` name
 - Move annotations from field to getter
 - Recalculate `properGetterName` by the field name to `getId` and overwrite method name to `properGetterName` (same for setter)

Either, I don't understand what the code is doing, which is quite possible due to missing tests to prove it, or the code is working only partially and needs revisiting.